sensor/common/sensor/central_sender_impl.go

```diff
@@ -112,9 +110,20 @@ func (s *centralSenderImpl) send(stream central.SensorService_CommunicateClient,
 			}
 
 			if msg.GetEvent().GetSynced() != nil {
-				log.Info("Sending synced signal to Central")
-				if s.onSync != nil {
-					go s.onSync()
+				unchangedIds := s.observationSet.Close()
+				// Enhance sync with all the observed IDs
+				msg.GetEvent().GetSynced().UnchangedIds = unchangedIds
```

Comment:
This is a strange usage of the getters. It's cleaner if you modify this in the if above as in

```
if syncEvent := msg.GetEvent().GetSynced(); syncEvent != nil 
```

then you operate on syncEvent

Reply:
Done